zio-http/src/main/scala/zhttp/service/RequestBodyHandler.scala

```diff
@@ -2,16 +2,23 @@ package zhttp.service
 import io.netty.channel.ChannelHandler.Sharable
 import io.netty.channel.{ChannelHandlerContext, SimpleChannelInboundHandler}
 import io.netty.handler.codec.http.HttpContent
+import zhttp.http.{Response, Status}
 import zhttp.service.server.content.handlers.UnsafeRequestHandler.{UnsafeChannel, UnsafeContent}
 
 @Sharable
-final case class RequestBodyHandler(msgCallback: (UnsafeChannel, UnsafeContent) => Unit)
-    extends SimpleChannelInboundHandler[Any](true) {
+final case class RequestBodyHandler[R](
+  msgCallback: (UnsafeChannel, UnsafeContent) => Unit,
+  config: Server.Config[R, Throwable],
+) extends SimpleChannelInboundHandler[Any](true) {
 
   override def channelRead0(ctx: ChannelHandlerContext, msg: Any): Unit = {
 
     if (msg.isInstanceOf[HttpContent]) {
-      msgCallback(UnsafeChannel(ctx), UnsafeContent(msg.asInstanceOf[HttpContent]))
+      val httpContent = msg.asInstanceOf[HttpContent]
+      if (httpContent.content().readableBytes() > config.maxRequestSize)
```

Comment:
condition is not correct, config should be passed to the unsafe body and should be handled there